modin/data_management/query_compiler/base_query_compiler.py

```diff
@@ -0,0 +1,1121 @@
+
+    # END UDF
+
+    # Manual Partitioning methods (e.g. merge, groupby)
+    # These methods require some sort of manual partitioning due to their
+    # nature. They require certain data to exist on the same partition, and
+    # after the shuffle, there should be only a local map required.
+    def groupby_agg(self, by, axis, agg_func, groupby_args, agg_args):
+        raise NotImplementedError("Must be implemented in children classes")
+
+    # END Manual Partitioning methods
+
+    def get_dummies(self, columns, **kwargs):
+        """Convert categorical variables to dummy variables for certain columns.
+
+        Args:
+            columns: The columns to convert.
+
+        Returns:
+            A new PandasDataManager.
+        """
+        raise NotImplementedError("Must be implemented in children classes")
+
+    # Indexing
+    def view(self, index=None, columns=None):
+        raise NotImplementedError("Must be implemented in children classes")
+
+    def squeeze(self, ndim=0, axis=None):
+        raise NotImplementedError("Must be implemented in children classes")
+
+    def write_items(self, row_numeric_index, col_numeric_index, broadcasted_items):
+        raise NotImplementedError("Must be implemented in children classes")
+
+    def global_idx_to_numeric_idx(self, axis, indices):
+        """
+        Note: this function involves making copies of the index in memory.
+
+        Args:
+            axis: Axis to extract indices.
+            indices: Indices to convert to numerical.
+
+        Returns:
+            An Index object.
+        """
+        raise NotImplementedError("Must be implemented in children classes")
+
+    def enlarge_partitions(self, new_row_labels=None, new_col_labels=None):
+        raise NotImplementedError("Must be implemented in children classes")
+
+    # END Abstract methods for QueryCompiler
+
+    @property
+    def __constructor__(self):
+        """By default, constructor method will invoke an init."""
+        return type(self)
+
+    # Append/Concat/Join (Not Merge)
+    # The append/concat/join operations should ideally never trigger remote
+    # compute. These operations should only ever be manipulations of the
+    # metadata of the resulting object. It should just be a simple matter of
+    # appending the other object's blocks and adding np.nan columns for the new
+    # columns, if needed. If new columns are added, some compute may be
+    # required, though it can be delayed.
+    #
+    # Currently this computation is not delayed, and it may make a copy of the
+    # DataFrame in memory. This can be problematic and should be fixed in the
+    # future. TODO (devin-petersohn): Delay reindexing
+    def join(self, other, **kwargs):
+        """Joins a list or two objects together.
+
+        Args:
+            other: The other object(s) to join on.
+
+        Returns:
+            Joined objects.
+        """
+        if not isinstance(other, list):
```

Comment:
Let's just leave this one abstract and have the implementation details figured out in the children classes.